### ISSUE TITLE
expected NoMethodError with message matching /undefined method `no_such_method'

### ISSUE THREAD

**Issue by digitalextremist** [celluloid/celluloid#560]
```
1) Celluloid::SyncCall when obj does not respond to a method when obj raises during inspect should emulate obj.inspect
     Failure/Error: expect { actor.no_such_method }.to raise_exception(
       expected NoMethodError with message matching /undefined method `no_such_method' for #\<CallExampleActor:0x[a-f0-9]+ @next=nil>/, got #<RSpec::Mocks::MockExpectationError: (#<CallExampleActor:0x762c5469>).inspect(no args)
           expected: 0 times with any arguments
           received: 1 time> with backtrace:
         # ./lib/celluloid/calls.rb:57:in `simulated_inspect_dump'
         # ./lib/celluloid/calls.rb:55:in `simulated_inspect_dump'
         # ./lib/celluloid/calls.rb:37:in `check'
         # ./lib/celluloid/calls.rb:35:in `check'
         # ./lib/celluloid/calls.rb:32:in `check'
         # ./lib/celluloid/calls.rb:24:in `dispatch'
         # ./lib/celluloid/calls.rb:79:in `dispatch'
         # ./lib/celluloid/cell.rb:60:in `invoke'
         # ./lib/celluloid/cell.rb:71:in `task'
         # ./lib/celluloid/actor.rb:357:in `task'
         # ./lib/celluloid/tasks.rb:57:in `initialize'
         # ./lib/celluloid/tasks.rb:47:in `initialize'
         # ./lib/celluloid/tasks/fibered.rb:15:in `create'
         # (celluloid):0:in `remote procedure call'
         # ./lib/celluloid/calls.rb:108:in `value'
         # ./lib/celluloid/proxies/sync_proxy.rb:37:in `method_missing'
         # ./spec/celluloid/calls_spec.rb:34:in `(root)'
         # ./spec/celluloid/calls_spec.rb:34:in `(root)'
         # ./spec/spec_helper.rb:92:in `(root)'
         # ./spec/spec_helper.rb:70:in `(root)'
         # ./spec/spec_helper.rb:65:in `(root)'
     # ./spec/celluloid/calls_spec.rb:34:in `(root)'
     # ./spec/spec_helper.rb:92:in `(root)'
     # ./spec/spec_helper.rb:70:in `(root)'
     # ./spec/spec_helper.rb:65:in `(root)'
```

**Comment by e2** [comment]
@digitalextremist - I'll work this out. It's up next.

**Comment by digitalextremist** [comment]
:ok_hand:

**Comment by e2** [comment]
A fix is in #559. 

The cause: - the recently enabled sync calls mean that it's safe to call `:inspect`, and JRuby mocking is slightly different - it fired. This means the test wasn't 100% correct.

**Comment by digitalextremist** [comment]
Verified resolved in current `0.17.0` as of now, for now.
